Apps/Playground/CMakeLists.txt

```diff
@@ -0,0 +1,171 @@
+
+if(WINDOWS_STORE)
+    target_compile_options(Playground PRIVATE /ZW)
+    target_compile_options(Playground PRIVATE /await)
+endif()
+
+if(WIN32)
+    set(TEST_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../Library/Dependencies/bgfx.cmake/bx/include/compat/msvc"
+        "${CMAKE_CURRENT_SOURCE_DIR}/../../Library/Dependencies/bgfx.cmake/bgfx/include"
+        "${CMAKE_CURRENT_SOURCE_DIR}/../../Library/Dependencies/bgfx.cmake/bx/include"
+        "${CMAKE_CURRENT_SOURCE_DIR}/../../Library/Dependencies/bgfx.cmake/bimg/include")
+endif()
+
+target_include_directories(Playground PRIVATE "Source" "." ${TEST_DEPENDENCY})
+
+target_link_libraries(Playground PRIVATE Library ${ADDITIONAL_LIBRARIES})
+
+if(APPLE)
+    if(IOS)
+        set_target_properties(Playground PROPERTIES
+            MACOSX_BUNDLE true
+            MACOSX_BUNDLE_INFO_PLIST "${PLIST_FILE}"
+            XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES
+            RESOURCE "${RESOURCE_FILES}"
+
+            XCODE_ATTRIBUTE_IPHONEOS_DEPLOYMENT_TARGET 9.0
+            XCODE_ATTRIBUTE_PRODUCT_BUNDLE_IDENTIFIER "com.BabylonNative.Playground.iOS"
+
+            XCODE_ATTRIBUTE_ENABLE_BITCODE NO
+
+            XCODE_ATTRIBUTE_SWIFT_VERSION "4.0"
+            XCODE_ATTRIBUTE_SWIFT_OBJC_BRIDGING_HEADER "${CMAKE_CURRENT_LIST_DIR}/iOS/LibNativeBridge.h"
+            XCODE_ATTRIBUTE_LD_RUNPATH_SEARCH_PATHS "@executable_path/Frameworks"
+            XCODE_ATTRIBUTE_ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES YES
+
+            # CMake seems to add a custom flag "-Wno-unknown-pragmas" to the Swift compiler. That flag is used for Clang,
+            # So we need to make sure we override it with nothing here in order to compile Swift.
+            XCODE_ATTRIBUTE_OTHER_SWIFT_FLAGS "")
+
+            # Swift support
+            set(CMAKE_Swift_COMPILER_FORCED TRUE)
+            set(CMAKE_Swift_LANGUAGE_VERSION 4.0)
+            enable_language(Swift)
+    else()
+        set_target_properties(Playground PROPERTIES
+            MACOSX_BUNDLE true
+            MACOSX_BUNDLE_INFO_PLIST "${PLIST_FILE}"
+            XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES
+            RESOURCE "${RESOURCE_FILES}"
+            FOLDER "Playground")
+    endif()
+endif()
+
+if(WINDOWS_STORE)
+    set_property(SOURCE ${SCRIPTS} ${BABYLONJS} PROPERTY VS_DEPLOYMENT_CONTENT 1)
+    set_property(SOURCE ${SCRIPTS} ${BABYLONJS} PROPERTY VS_DEPLOYMENT_LOCATION "Scripts")
+else()
+    foreach(script ${SCRIPTS})
+        add_custom_command(
+            OUTPUT "${script}"
+            COMMAND "${CMAKE_COMMAND}" -E copy "${CMAKE_CURRENT_SOURCE_DIR}/${script}" "${CMAKE_CURRENT_BINARY_DIR}/${script}"
+            COMMENT "Copying ${script}"
+            MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/${script}")
+    endforeach()
+    
+    add_custom_command(TARGET Playground PRE_BUILD
+                   COMMAND ${CMAKE_COMMAND} -E copy_directory
+                   ${CMAKE_SOURCE_DIR}/Apps/Babylonjs/ "${CMAKE_CURRENT_BINARY_DIR}/Scripts")
```

Comment:
I can't remember if this works right or not, but I seem to remember `copy_directory` doesn't work well if only one file changed.

Reply:
Replaced with a file copy

Reply:
Should use `copy_if_different`.